### PR TITLE
Fix broken links and redirects

### PR DESCRIPTION
**Pull request by dcdunkan**
Fixed few broken links and updated the redirected links. Need to check the relative links as well.

Related:
- https://github.com/grammyjs/grammY/issues/425
- https://github.com/grammyjs/link-checker

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
Did you use a script for this? Can we run it in CI?

**Comment by roj1512** [comment]
Give me your tech @dcdunkan

**Comment by dcdunkan** [comment]
I used a script. It not that good though. I can share it after I add some functionalities to check the relative (local) links.

**Comment by roj1512** [comment]
Alright, waiting for that :)

**Comment by dcdunkan** [comment]
Might take some time! :D

**Comment by roj1512** [comment]
Take it!

**Comment by KnorpelSenf** [comment]
And please don't hesitate to share bad, broken code because that will only make it better :)

**Comment by dcdunkan** [comment]
https://gist.github.com/dcdunkan/38725722806aca8012af485dfa283845 (comments are mostly myself)

Sorry, it took me some time to finally finish the script. I don't know if its much helpful in CI. I re-wrote the original script hoping that it would make it better, and it turns out that the rewrite isn't better either. So, here I am, sharing the bad code without hesitation @KnorpelSenf :)

(There could be bugs that I haven't noticed.)

Tried to add more features, e.g. showing the line numbers with the issue; but failed to finish it, I couldn't get all the line numbers. It is still possible though. Its just that I was confused by my own script :D

Currently the script,
- fetches the external links and take care of redirections and 404s.
- resolves local files and make sure that the file exists.
- gets anchor IDs from a local file or website and makes sure that all the linked anchors are still there.

This script can be ran before building the website as I didn't think and try to find the broken links with a built website. If its supposed to be ran after building the website, with a few changes, the script can take of that as well.

A part of the output:

image

and there's a tiny summary at the end:

image

**Comment by roj1512** [comment]
Awesome work! A few changes and then we will be using it here ;)

**Comment by dcdunkan** [comment]
Fixed all the redirections other than for a https://localtunnel.me. Would like your opinion on that. Should I replace it with the redirect URL (Github pages)? The confusion is because their GitHub repository also links https://localtunnel.me as the main link.

All that left is to fix the ~50 anchor links; all local files.

File: site/docs/zh/plugins/i18n.md [hunk 1]
```diff
@@ -61,7 +61,7 @@
 ### 你为什么需要这个插件？
 
 这个插件可以在国际化的过程中帮助你。
-它基于 [Fluent](https://projectfluent.org/)——一个由 [Mozilla](https://mozilla.org/en-US/) 开发的本地化系统。
+它基于 [Fluent](https://projectfluent.org/) ——一个由 [Mozilla](https://mozilla.org/) 开发的本地化系统。
```
[the remark this answers is not in the document]
Comment (roj1512) [reply site/docs/zh/plugins/i18n.md]: Good question.

[the remark this answers is not in the document]
Comment (dcdunkan) [reply site/docs/zh/plugins/i18n.md]: I don't think that space was added as a part of this pull request. It belongs to the commit made to `main` here: https://github.com/grammyjs/website/commit/79aae7f71d294ec5e5e07199bfb514f7b3e197fd. It got here when this branch was synced with `main`.

**Comment by dcdunkan** [comment]
I would appreciate if the Chinese translators can fix the rest of the Chinese anchors. It's a little hard to find the correct anchors since I don't know Chinese and the characters are hard to differentiate. @roj1512 Could you ping those who currently works on the Chinese translation?

**Comment by roj1512** [comment]
Pinged.

**Comment by KnorpelSenf** [comment]
@dcdunkan is there anything left to do here now, besides resolving the conflicts?

**Comment by dcdunkan** [comment]
image

Nope. I fixed the last few redirection errors in the link-checker itself. This is ready for reviewing now.

**Review by KnorpelSenf — APPROVED** [review]
Impressive results, we should've had this earlier

**Pull request merged**